### PR TITLE
#166324980 Creates fines model

### PR DESCRIPTION
**Pull request by Lundii**
#### What does this PR do?
It creates a model for storing fines
#### Description of Task to be completed?
* created a migration file for the fines model
#### How should this be manually tested?
* git pull or fetch this branch
* run npm install to get all dependencies
* run npm run db:roll to create the model
* confirm the model is created in you db
#### Any background context you want to provide?
No
#### What are the relevant pivotal tracker stories?
[#166324980](https://www.pivotaltracker.com/story/show/166324980)
#### Screenshots (if appropriate)
No
#### Questions:
No

### CODE REVIEW

**Review by henryade — CHANGES_REQUESTED** [review]
I cant see price in your models file.

**Comment by Lundii** [comment]
> I cant see price in your models file.

Done

**Review by allebd — CHANGES_REQUESTED** [review]
No association of fines to users in the user model

No association of fines to book in the book model

**Comment by Lundii** [comment]
> No association of fines to users in the user model
> 
> No association of fines to book in the book model

Done

File: server/database/models/books.js [hunk 1]
```diff
@@ -16,6 +16,10 @@ module.exports = (sequelize, DataTypes) => {
       foreignKey: 'authorId',
       onDelete: 'CASCADE'
     });
+
+    Books.hasMany(models.Fines, {
+      as: 'id'
```
Comment (Ukhu) [inline server/database/models/books.js]: Remove the 'as' property. Just specify the foreign key and the onDelete constraint here.

File: server/database/models/fines.js [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+    bookId: DataTypes.INTEGER,
+    price: DataTypes.INTEGER,
+    status: DataTypes.STRING,
+    description: DataTypes.STRING,
+    type: DataTypes.STRING
+  }, {});
+  Fines.associate = function(models) {
+    // associations can be defined here
+    Fines.belongsTo(models.Users, {
+      foreignKey: 'id',
```
Comment (Ukhu) [inline server/database/models/fines.js]: The 'foreignKey' here is the 'userId' not 'id'.

File: server/database/models/fines.js [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+      sourceKey: 'userId',
```
Comment (Ukhu) [inline server/database/models/fines.js]: No need for 'sourceKey' here

File: server/database/models/fines.js [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+      onDelete: 'CASCADE'
+    });
+
+    Fines.belongsTo(models.Books, {
+      foreignKey: 'id',
```
Comment (Ukhu) [inline server/database/models/fines.js]: The 'foreignKey' here is the 'bookId' not 'id'.

File: server/database/models/fines.js [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+      sourceKey: 'bookId',
```
Comment (Ukhu) [inline server/database/models/fines.js]: No need for 'sourceKey' here

File: server/database/models/users.js [hunk 6]
```diff
@@ -15,6 +15,9 @@ module.exports = (sequelize, DataTypes) => {
       foreignKey: 'userId',
       onDelete: 'CASCADE'
     });
+    Users.hasMany(models.Fines, {
+      as: 'id'
```
Comment (Ukhu) [inline server/database/models/users.js]: Specify the 'foreignKey' and the 'onDelete' constraint. No need for 'as' here.

**Review by Ukhu — CHANGES_REQUESTED** [review]
Rebase branch

File: server/database/migrations/20190530001212-create-fines.js [hunk 7]
```diff
@@ -0,0 +1,57 @@
...
+  up: (queryInterface, Sequelize) => {
+    return queryInterface.createTable('Fines', {
+      id: {
+        allowNull: false,
+        autoIncrement: true,
+        primaryKey: true,
+        type: Sequelize.INTEGER
+      },
+      userId: {
+        allowNull: false,
```
Comment (Ukhu) [inline server/database/migrations/20190530001212-create-fines.js]: Add 'onDelete' constraint.

File: server/database/migrations/20190530001212-create-fines.js [hunk 8]
```diff
@@ -0,0 +1,57 @@
...
+        type: Sequelize.INTEGER,
+        references: {
+          model: 'Users',
+          key: 'id'
+        }
+      },
+      bookId: {
+        allowNull: false,
```
Comment (Ukhu) [inline server/database/migrations/20190530001212-create-fines.js]: Add 'onDelete' constraint.

File: server/database/models/users.js [hunk 9]
```diff
@@ -25,6 +25,10 @@ module.exports = (sequelize, DataTypes) => {
       foreignKey: 'userId',
       onDelete: 'CASCADE'
     });
```
Comment (Ukhu) [inline server/database/models/users.js]: Add spacing here.

**Review by Ukhu — DISMISSED** [review]

**Comment by allebd** [comment]
Kindly work on your commit message, it is not following the right convention, don't forget to use imperative and present tense on your header and body message

**Comment by Lundii** [comment]
> Kindly work on your commit message, it is not following the right convention, don't forget to use imperative and present tense on your header and body message

Done

**Review by Ukhu — DISMISSED** [review]

**Review by henryade — DISMISSED** [review]

**Review by Nennyfills — DISMISSED** [review]
Nice

**Review by KvNGCzA — CHANGES_REQUESTED** [review]
There are issues with your associations. Please look into them

**Comment by Lundii** [comment]
> There are issues with your associations. Please look into them

Fixed

**Pull request merged**